crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -196,15 +196,19 @@ impl Deref for BindGroup {
 ///
 /// ## `storage(BINDING_INDEX, arguments)`
 ///
-/// * The field's [`Handle<Storage>`](bevy_asset::Handle) will be used to look up the matching [`Buffer`] GPU resource, which
-///   will be bound as a storage buffer in shaders. If the `storage` attribute is used, the field is expected a raw
-///   buffer, and the buffer will be bound as a storage buffer in shaders.
-///
-/// | Arguments              | Values                                                                  | Default              |
-/// |------------------------|-------------------------------------------------------------------------|----------------------|
-/// | `visibility(...)`      | `all`, `none`, or a list-combination of `vertex`, `fragment`, `compute` | `vertex`, `fragment` |
-/// | `read_only`            | if present then value is true, otherwise false                          | `false`              |
-/// | `buffer`               | if present then the field will be assumed to be a raw wgpu buffer       |                      |
+/// * The field's [`Handle<Storage>`](bevy_asset::Handle) will be used to look
+///   up the matching [`Buffer`] GPU resource, which will be bound as a storage
+///   buffer in shaders. If the `storage` attribute is used, the field is expected
+///   a raw buffer, and the buffer will be bound as a storage buffer in shaders.
+///   In bindless mode, `binding_array()` argument that specifies the binding
+///   number of the resulting storage buffer binding array must be present.
+///
+/// | Arguments              | Values                                                                  | Default                |
+/// |------------------------|-------------------------------------------------------------------------|------------------------|
+/// | `visibility(...)`      | `all`, `none`, or a list-combination of `vertex`, `fragment`, `compute` | `vertex`, `fragment`   |
+/// | `read_only`            | if present then value is true, otherwise false                          | `false`                |
+/// | `buffer`               | if present then the field will be assumed to be a raw wgpu buffer       |                        |
+/// | `binding_array(...)`   | the binding number of the binding array, for bindless mode              | bindless mode disabled |
```

Comment:
I don't get what this means. Is the binding number like `@group(...) @binding(X) var<storage> binding_array<T>;` X here?

```suggestion
/// | `binding_array(...)`   | the binding number of the binding array, for bindless mode              | bindless mode disabled |
```

Reply:
Yes, exactly. When you write something like `#[storage(10, binding_array(11))]`, the corresponding WGSL declaration looks like `@group(2) @binding(11) var<storage> binding_array<T>;`. In this case, the 10 means that the index *into* that binding array is given by the 10th *field* of the bindings structure (e.g. `StandardMaterialBindings`).

Thus there are two indices: the *bindless index*, which refers to a field of the structure that contains all the indices into the binding arrays for a single material, and the *binding number*, which refers to the actual `@binding` in the bind group layout.